pkg/tests/tasks/security/authorization/ext_auth_test.go

```diff
@@ -107,6 +107,88 @@ spec:
 	})
 }
 
+func TestEnvoyExtAuthzGrpcExtensionProvider(t *testing.T) {
+	test.NewTest(t).Id("T42").Groups(test.Full, test.InterOp).Run(func(t test.TestHelper) {
+		if env.GetSMCPVersion().LessThan(version.SMCP_2_4) {
+			t.Skip("extensionProviders.envoyExtAuthzGrpc was added in v2.4")
+		}
+		t.Log("This test validates authorization policies with a JWT Token")
+
+		ns := "foo"
+
+		ossm.DeployControlPlane(t)
+
+		t.LogStep("Install httpbin and sleep")
+		app.InstallAndWaitReady(t, app.Httpbin(ns), app.Sleep(ns))
+		t.Cleanup(func() {
+			app.Uninstall(t, app.Httpbin(ns), app.Sleep(ns))
+		})
+
+		t.LogStep("Check if httpbin returns 200 OK when no authorization policies are in place")
+		assertHttpbinRequestSucceeds(t, ns, httpbinRequest("GET", "/ip"))
+
+		t.LogStep("Deploy the External Authorizer and Verify the sample external authorizer is up and running")
+		oc.ApplyTemplate(t, ns, ExternalAuthzService, nil)
+		t.Cleanup(func() {
+			oc.DeleteFromTemplate(t, ns, ExternalAuthzService, nil)
+		})
+
+		oc.WaitDeploymentRolloutComplete(t, ns, "ext-authz")
+
+		t.LogStep("Set envoyExtAuthzgRPC extension provider in SMCP")
+		if env.GetSMCPVersion().LessThan(version.SMCP_2_4) {
+			oc.Patch(t, meshNamespace, "smcp", smcpName, "merge", `
+spec:
+  techPreview:
+    meshConfig:
+      extensionProviders:
+      - name: sample-ext-authz-grpc
+        envoyExtAuthzGrpc:
+          includeRequestHeadersInCheck:
+          - x-ext-authz
+          port: "9000"
+          service: ext-authz.foo.svc.cluster.local`)
+
+			t.Cleanup(func() {
+				oc.Patch(t, meshNamespace, "smcp", smcpName, "json",
+					`[{"op": "remove", "path": "/spec/techPreview"}]`)
+			})
+
+		} else {
+			oc.Patch(t, meshNamespace, "smcp", smcpName, "merge", `
+spec:
+  meshConfig:
+    extensionProviders:
+    - name: sample-ext-authz-grpc
+      envoyExtAuthzGrpc:
+        includeRequestHeadersInCheck:
+        - x-ext-authz
+        port: 9000
+        service: ext-authz.foo.svc.cluster.local`)
```

Comment:
`envoyExtAuthzGrpc` is not yet supported in `spec.meshConfig.extensionProviders`.
```suggestion
```

Reply:
@jewertow Since it is only in techPreview, do you want me to take out this whole section?